pycbc/waveform/generator.py

```diff
@@ -671,16 +671,30 @@ def generate(self, **kwargs):
         h = {}
         if self.detector_names != ['RF']:
             for detname, det in self.detectors.items():
-                # apply detector response function
-                fp, fc = det.antenna_pattern(self.current_params['ra'],
-                            self.current_params['dec'],
-                            self.current_params['polarization'],
-                            self.current_params['tc'])
+                refframe = self.current_params.get('tc_ref_frame', 'geocentric')
+                ra = self.current_params['ra']
+                dec = self.current_params['dec']
+                ref_tc = self.current_params['tc']
+                pol = self.current_params['polarization']
+                # apply response function
+                fp, fc = det.antenna_pattern(ra, dec, pol, ref_tc)
                 thish = fp*hp + fc*hc
```

Comment:
The tc at the detector should be used for the antenna pattern rather than the `ref_tc` (this is actually a minor bug in current master). To do that, move lines 680 and 681 to line 697; i.e., just after the time shift conditional block, but before `apply_fd_time_shift` is run.